napari/_vispy/vendored/image.py

```diff
@@ -113,8 +113,8 @@
 _c2l = 'float cmap(vec4 color) { return (color.r + color.g + color.b) / 3.; }'
 
 
-def _build_color_transform(data, clim, gamma, cmap):
-    if data.ndim == 2 or data.shape[2] == 1:
+def _build_color_transform(grayscale: bool, clim, gamma, cmap):
```

Comment:
I'm ok with this change, it's in a private method that only gets used once, inside the file. I actually find it much more readable with `grayscale` too. We've got another PR #1853 that touches this too, but in a different way, so I think that's fine too. It's also clear how to adjust #1853 to solve the merge conflict.

We will need to push this up to vispy one day most likely, but I think it should be fine as the method is private

Reply:
Yeah the next step on the vispy-side would be take all those private methods and put them in a file like `image_util.py` so derived classes can easily call into them without having to call into private methods in `image.py`.

They are only private because no one else needed them. But if derived image-related classes want to use them as well, there's no downside in splitting them out, that I see.

How have to copy and modify `_prepare_draw()` is also not great. But less obvious how to fix that up. But ideally `ImageVisual` can be extended with derived classes in a nice way without duplication or reaching into private methods.

But that's why you have to refactor base classes once you've made some derived classes and seen the pain points.